data/moves.js

```diff
@@ -13516,7 +13516,11 @@ let BattleMovedex = {
 		flags: {snatch: 1, heal: 1},
 		onTryMove(pokemon) {
 			if (pokemon.hp < pokemon.maxhp && pokemon.status !== 'slp' && !pokemon.hasAbility('comatose')) return;
-			this.add('-fail', pokemon);
+			if (pokemon.hp === pokemon.maxhp && pokemon.status) {
+				this.add('-message', pokemon.name + "'s HP is full!");
```

Comment:
In other words, this should be:

```suggestion
				this.add('-fail', pokemon, 'move: Rest');
```

Reply:
Oh there was a fail message for rest added to the client? Good to know

Reply:
No, but you (or someone else) should add it to the client.

Reply:
ah

Reply:
i'll look at the client and see if i can figure out how to do that change. shouldn't be too difficult i'd imagine

Reply:
You just need to add

```
fail: "[POKEMON]'s HP is full!",
```

to `data/text.js`.

Reply:
oh, okay thanks. that makes it easier since i'd have to have found the correct file